src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -49,14 +48,48 @@ internal static IEnumerable<CompletionResult> GetMatchingResults(
                 if (matchStrategy(value, wordToComplete))
                 {
                     string completionText = QuoteCompletionText(value, quote);
-                    string toolTip = toolTipMapping?.Invoke(value) ?? value;
-                    string listItemText = listItemTextMapping?.Invoke(value) ?? value;
+
+                    CompletionDisplayInfo displayInfo = displayInfoMapper(value);
+                    string toolTip = displayInfo.ToolTip;
+                    string listItemText = displayInfo.ListItemText;
 
                     yield return new CompletionResult(completionText, listItemText, resultType, toolTip);
                 }
             }
         }
 
+        /// <summary>
+        /// Represents the display information for a completion result.
+        /// </summary>
+        internal class CompletionDisplayInfo
```

Comment:
How will users create this in their code and scripts? We will have to make this class public. Can we avoid this? Can we come up with something more convenient for them? Maybe simple tuple?

Reply:
yeah was thinking about this, a tuple does work as well but usually classes/records are preferred.

I am happy to use tuple if you see no issues with it, given its always going to be two elements in this anyways and unlikely to grow.

Reply:
Changed to tuple here: f0c21a2e05ad37c588ff03543f63a14c5db8e9f0